### PR TITLE
fix: Pydantic and PR Target Issue

### PR DESCRIPTION
**Pull request by actuarysailor**
@andrewthetechie - this is my last PR to your repo; I will self maintain going forward.  These address the issues with forks submitting changes and PR Target using main instead of the PR branch.  If you implement this, it should use the PR branch going forward and avoid Token differences with the normal PR (because PR Target fixes that).

### CODE REVIEW

File: .github/workflows/action-integration.yml [hunk 1]
```diff
@@ -4,14 +4,16 @@ on:
...
   action-integration-testing:
     name: Action Integration Testing
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-        name: Checkout
+      - name: Checkout
+        uses: actions/checkout@v4.1.6
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (andrewthetechie) [inline .github/workflows/action-integration.yml:16]: I just tested this and this breaks any running of this test on non-prs. This test runs on every push to main too.

Comment (actuarysailor) [reply .github/workflows/action-integration.yml:16]: You could add a coalesce with GitHub.ref

**Comment by actuarysailor** [comment]
@andrewthetechie  - what does this action do?  I keep getting this error now that I am trying to maintain my own repo. 
https://github.com/actuarysailor/gha-repo-manager/actions/runs/9341793728

**Comment by actuarysailor** [comment]
@andrewthetechie  - ok, finally got everything working on my end, so I will self-maintain going forward.

https://github.com/marketplace/actions/manage-administrative-repository-settings-from-pull-requests

**Comment by actuarysailor** [comment]
@andrewthetechie - all the workflows are working in my repo, but when I reference remotely I get this error:

Error response from daemon: Head "https://ghcr.io/v2/actuarysailor/gha-repo-manager/manifests/v1.9.0": unauthorized
Warning: Docker pull failed with exit code 1, back off 5.[7](https://git.nylcloud.com/Actuarial-Services/.github/actions/runs/109690/jobs/236114#step:3:7)94 seconds before retry.
/usr/local/bin/docker pull ghcr.io/actuarysailor/gha-repo-manager:v1.9.0
Error response from daemon: Head "https://ghcr.io/v2/actuarysailor/gha-repo-manager/manifests/v1.[9](https://git.nylcloud.com/Actuarial-Services/.github/actions/runs/109690/jobs/236114#step:3:9).0": unauthorized

**Pull request closed without merging**